python-package/lightgbm/engine.py

```diff
@@ -733,11 +788,10 @@ def cv(
     if callable(params["objective"]):
         fobj = params["objective"]
         params["objective"] = "none"
-    for alias in _ConfigAliases.get("num_iterations"):
-        if alias in params:
-            _log_warning(f"Found '{alias}' in params. Will use it instead of 'num_boost_round' argument")
-            num_boost_round = params.pop(alias)
-    params["num_iterations"] = num_boost_round
+
+    params = _choose_num_iterations(num_boost_round_kwarg=num_boost_round, params=params)
+    num_boost_round = params["num_iterations"]
```

Comment:
Is it OK to leave `num_iterations` key in params and pass them to cpp side, instead of pop-ing it like before?

Reply:
It was already being passed to the cpp side... the result of all the processing that resolves keyword arguments and `params` entries is assigned back to `params["num_iterations"]` right here: https://github.com/microsoft/LightGBM/blob/d67ecf94259217746b031a945d66f4e0fcec8579/python-package/lightgbm/engine.py#L205

So I think this PR does not change the behavior of what is passed through to the C++ side.

Reply:
Ah, indeed! Overlooked this moment, sorry.